.github/workflows/avm.platform.publish-module-index-json.yml

```
@@ -0,0 +1,77 @@
+
+      - name: Log in to Azure
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.PUBLISH_CLIENT_ID }}
+          tenant-id: ${{ secrets.PUBLISH_TENANT_ID }}
+          subscription-id: ${{ secrets.PUBLISH_SUBSCRIPTION_ID }}
+          enable-AzPSSession: true
+
+      - name: Install Azure Powershell Modules
+        shell: pwsh
+        run: |
+          Install-Module -Name Az.Storage -Force
+
+      - name: Generate moduleIndex.json
+        shell: pwsh
+        run: |
+          # Load used functions
+          . (Join-Path $env:GITHUB_WORKSPACE 'avm' 'utilities' 'pipelines' 'platform' 'Invoke-AvmJsonModuleIndexGeneration.ps1')
+
+          $functionInput = @{
+            storageAccountName                           = 'biceplivedatasaprod'
+            storageAccountContainer                      = 'bicep-cdn-live-data-container'
+            storageBlobName                              = 'module-index'
+            moduleIndexJsonFilePath                      = 'moduleIndex.json'
+            prefixForLastModuleIndexJsonFile             = 'last-'
+            prefixForCurrentGeneratedModuleIndexJsonFile = 'generated-'
+            doNotMergeWithLastModuleIndexJsonFileVersion = $false
+          }
+
+          Write-Verbose "Invoke task with" -Verbose
+          Write-Verbose ($functionInput | ConvertTo-Json | Out-String) -Verbose
+
+          Invoke-AvmJsonModuleIndexGeneration @functionInput
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: publish-module-index-json-artifacts
+          path: |
+            moduleIndex.json
+            last-moduleIndex.json
+            generated-moduleIndex.json
+
+      - name: Upload to blob storage
+        shell: pwsh
+        run: |
+          $storageAccountInfo = @{
+            storageAccountName      = 'biceplivedatasaprod'
+            storageAccountContainer = 'bicep-cdn-live-data-container'
+            storageBlobName         = 'module-index'
+            storageBlobContentType  = @{"ContentType" = "application/json"}
+          }
+
+          Write-Verbose "Uploading moduleIndex.json to blob storage account $storageAccountInfo.storageAccountName in container $storageAccountInfo.storageAccountContainer as blob $storageAccountInfo.storageBlobName" -Verbose
+          $storageContext = New-AzStorageContext -StorageAccountName $storageAccountInfo.storageAccountName -UseConnectedAccount
+          Set-AzStorageBlobContent -Context $storageContext -Container $storageAccountInfo.storageAccountContainer -Blob $storageAccountInfo.storageBlobName -File 'moduleIndex.json' -Properties $storageAccountInfo.storageBlobContentType  -Force
+          Write-Verbose "Uploaded moduleIndex.json to blob storage account $storageAccountInfo.storageAccountName in container $storageAccountInfo.storageAccountContainer as blob $storageAccountInfo.storageBlobName" -Verbose
```

Fairly certain, the verbose messages did not work. 
Fairly certain the verbose messages did not work. 

```suggestion
          Write-Verbose ('Uploading [moduleIndex.json] to blob storage account [{0}] in container [{1}] as blob [{2}]' -f $storageAccountInfo.storageAccountName, $storageAccountInfo.storageAccountContainer, $storageAccountInfo.storageBlobName) -Verbose
          
          $storageContext = New-AzStorageContext -StorageAccountName $storageAccountInfo.storageAccountName -UseConnectedAccount
          
          $functionInput = @{
              Context    = $storageContext 
              Container  = $storageAccountInfo.storageAccountContainer 
              Blob       = $storageAccountInfo.storageBlobName 
              File       = 'moduleIndex.json' 
              Properties = $storageAccountInfo.storageBlobContentType  
          }
          Set-AzStorageBlobContent @functionInput -Force
          
          Write-Verbose ('Upload of [{0}] complete.' -f $storageAccountInfo.storageBlobName) -Verbose
```